### ISSUE TITLE
Can't get L011 to pass when aliasing is set to implicit

### ISSUE THREAD

**Issue by nickbayley** [sqlfluff/sqlfluff#2682]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

When rule L011 is set to `implicit`, I get an L011 violation regardless of whether I include an explicit `as` or not.

`select foo.bar from some_table foo;` causes an L011 violation.

As does `select foo.bar from some_table as foo;`. Although this is what I would expect.

Maybe I am misunderstanding something? Sorry if that is the case. I also tried this in an older version (`0.6.6` - when this config option was introduced) and I have the same problem.

I normally use `oracle`, so aliasing in the from clause is forbidden (afaik). How should rules like L011 be approached when an option (in this case `explicit`) is not supported?

### Expected Behaviour

In my use case, no L011 violation should take place.

### Observed Behaviour

```
11:13:12 ❯ sqlfluff lint test.sql -vv
==== sqlfluff ====
sqlfluff:               0.10.1 python:                 3.10.2
implementation:        cpython dialect:                  ansi
verbosity:                   2 templater:               jinja

== Raw Config:
core:                                   
    dialect:            ansi                
    disable_noqa:       False               
    encoding:           autodetect          
    exclude_rules:      L031                
    fix_even_unparsable:False               
    ignore:             []                  
    ignore_templated_areas:True                
    nocolor:            False               
    output_line_length: 80                  
    recurse:            True                
    runaway_limit:      10                  
    sql_file_exts:      .sql,.sql.j2,.dml,.ddl
    templater:          jinja               
    verbose:            2                   
indentation:                            
    indented_ctes:      False               
    indented_joins:     False               
    indented_using_on:  True                
    template_blocks_indent:True                
rules:                                  
    allow_scalar:       True                
    comma_style:        trailing            
    indent_unit:        space               
    max_line_length:    80                  
    single_table_references:consistent          
    tab_space_size:     4                   
    unquoted_identifiers_policy:all                 
    L007:                                   
        operator_new_lines: after               
    L010:                                   
        capitalisation_policy:consistent          
    L011:                                   
        aliasing:           implicit            
    L012:                                   
        aliasing:           explicit            
    L014:                                   
        extended_capitalisation_policy:consistent          
    L016:                                   
        ignore_comment_clauses:False               
        ignore_comment_lines:False               
    L026:                                   
        force_enable:       False               
    L028:                                   
        force_enable:       False               
    L029:                                   
        unquoted_identifiers_policy:aliases             
    L030:                                   
        capitalisation_policy:consistent          
    L038:                                   
        select_clause_trailing_comma:forbid              
    L040:                                   
        capitalisation_policy:consistent          
    L042:                                   
        forbid_subquery_in: join                
    L047:                                   
        prefer_count_0:     False               
        prefer_count_1:     False               
    L052:                                   
        multiline_newline:  False               
        require_final_semicolon:False               
    L054:                                   
        group_by_and_order_by_style:consistent          
    L057:                                   
        additional_allowed_characters:""                  
        allow_space_in_identifier:False               
        quoted_identifiers_policy:all                 
        unquoted_identifiers_policy:all                 
    L059:                                   
        prefer_quoted_identifiers:False               
    L062:                                   
templater:                              
    unwrap_wrapped_queries:True                
    jinja:                                  
        apply_dbt_builtins: True                
        macros:                                 
            dbt_config:         {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
            dbt_is_incremental: {% macro is_incremental() %}True{% endmacro %}
            dbt_ref:            {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
            dbt_source:         {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
            dbt_var:            {% macro var(variable, default='') %}item{% endmacro %}

==== readout ====

=== [ path: test.sql ] ===

== [test.sql] LINTING                                                                                                                                                                                                                          
== [test.sql] FAIL                                                                                                                                                                                                                             
L:   1 | P:  32 | L011 | Implicit/explicit aliasing of table.
==== summary ====
files:             1 violations:        1
clean files:       0 unclean files:     1
avg per file:   1.00 unclean rate:   100%
status:         FAIL
All Finished 📜 🎉!
```

### How to reproduce

``` sql
-- ./test.sql
select foo.bar from some_table foo;
```

```
11:20:59 ❯ sqlfluff lint test.sql
== [test.sql] FAIL                                                                                                                                                                                                                             
L:   1 | P:  32 | L011 | Implicit/explicit aliasing of table.
All Finished 📜 🎉!
```


### Dialect

`ansi` / `oracle`

### Version

```
sqlfluff, version 0.10.1
Python 3.10.2
```

### Configuration

```
[sqlfluff]
exclude_rules = L031

[sqlfluff:rules:L011]
aliasing = implicit
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
This looks like a total bug. Not sure how no one ever spotted that before! Guess most people using implicit alias's just turn this rule off instead.

Should be fixed with #2683
